plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java

```diff
@@ -96,6 +97,17 @@ public void search(SearchRequest request, ActionListener<SearchResponse> actionL
                 Optional.ofNullable(request.source()).map(SearchSourceBuilder::fetchSource).map(FetchSourceContext::includes).orElse(null),
                 excludes.toArray(new String[0])
             );
+
+            // Check if the original query is not null before adding it to the must clause
```

Comment:
Is this a todo?

Reply:
no, it was the explanation why did I do that.

Reply:
Got it, this is a comment for line 103